compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/IrInlineCodegen.kt

```diff
@@ -158,6 +162,23 @@ class IrExpressionLambdaImpl(
     override val invokeMethodDescriptor: FunctionDescriptor = function.descriptor
 
     override val hasDispatchReceiver: Boolean = false
+
+    override fun generateLambdaBody(sourceCompiler: SourceCompilerForInline, reifiedTypeInliner: ReifiedTypeInliner) {
+        require(sourceCompiler is IrSourceCompilerForInline)
+        lateinit var methodNode: MethodNode
```

Comment:
Is it possible to move this logic and related one from ExpressionLambda to sourceCompiler? Seems all adapter related logic could be shared there

Reply:
Thanks, haven't noticed the similar code in doCreateMethodNodeFromSource